sdk/network/Azure.ResourceManager.Network/src/Generated/Models/EndpointServiceResult.cs

```diff
@@ -19,10 +19,17 @@ public EndpointServiceResult()
 
         /// <summary> Initializes a new instance of EndpointServiceResult. </summary>
         /// <param name="id"> Resource ID. </param>
-        /// <param name="name"> Resource name. </param>
-        /// <param name="resourceType"> Resource type. </param>
-        internal EndpointServiceResult(ResourceIdentifier id, string name, ResourceType? resourceType) : base(id, name, resourceType)
+        /// <param name="name"> Name of the endpoint service. </param>
+        /// <param name="resourceType"> Type of the endpoint service. </param>
+        internal EndpointServiceResult(ResourceIdentifier id, string name, string resourceType) : base(id)
         {
+            Name = name;
+            ResourceType = resourceType;
         }
+
+        /// <summary> Name of the endpoint service. </summary>
+        public string Name { get; }
+        /// <summary> Type of the endpoint service. </summary>
+        public string ResourceType { get; }
```

Comment:
Network has different combinations of the read-only attributes for `name` and `type` (id is always not read-only), this is the only case that both name and resourceType are read-only. I'm not going to create a XXResourceData base type just for this single case. We already have `NetworkResourceData` (name is not readonly, type is readonly), `NetworkWritableResourceData` (name and type are both not read-only) and they both have mutilple cases.